public/app/features/explore/LiveLogs.tsx

```diff
@@ -38,28 +38,98 @@ export interface Props extends Themeable {
   logsResult?: LogsModel;
   timeZone: TimeZone;
   stopLive: () => void;
+  onPause: () => void;
+  onResume: () => void;
+  isPaused: boolean;
 }
 
-class LiveLogs extends PureComponent<Props> {
+interface State {
+  logsResultToRender?: LogsModel;
+}
+
+class LiveLogs extends PureComponent<Props, State> {
   private liveEndDiv: HTMLDivElement = null;
+  private scrollContainerRef = React.createRef<HTMLDivElement>();
+  private lastScrollPos: number | null = null;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      logsResultToRender: props.logsResult,
+    };
+  }
 
   componentDidUpdate(prevProps: Props) {
-    if (this.liveEndDiv) {
-      this.liveEndDiv.scrollIntoView(false);
+    if (!prevProps.isPaused && this.props.isPaused) {
+      // So we paused the view and we changed the content size, but we want to keep the relative offset from the bottom.
+      if (this.lastScrollPos) {
+        // There is last scroll pos from when user scrolled up a bit so go to that position.
+        const { clientHeight, scrollHeight } = this.scrollContainerRef.current;
+        const scrollTop = scrollHeight - (this.lastScrollPos + clientHeight);
+        this.scrollContainerRef.current.scrollTo(0, scrollTop);
+        this.lastScrollPos = null;
+      } else {
+        // We do not have any position to jump to su the assumption is user just clicked pause. We can just scroll
+        // to the bottom.
+        if (this.liveEndDiv) {
+          this.liveEndDiv.scrollIntoView(false);
+        }
+      }
     }
   }
 
+  static getDerivedStateFromProps(nextProps: Props) {
```

Comment:
Does this best practice apply? 
https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization

Reply:
I think to some extent yes. We are not doing anything expensive here but it would be probably nicer to split this to 2 components and have this logic about what to actually show in a wrapper component, and have the other to only render it's props. Would leave it for separate PR thought as I do not think it creates any visible perf issue right now and would make this PR harder to follow.